crates/biome_js_analyze/src/lint/nursery/use_at_index.rs

```diff
@@ -0,0 +1,927 @@
+///     hoge[hoge.length - 0] // => None
+///     hoge[hoge.length - 1] // => Some(-1)
+///     hoge[fuga.length - 2] // => None
+/// ```
+fn extract_negative_index_expression(
+    member: &AnyJsExpression,
+    object: AnyJsExpression,
+) -> Option<AnyJsExpression> {
+    let (left, right_list) = get_left_node_from_minus_binary_expressions(member.clone())?;
+    if right_list.is_empty() {
+        return None;
+    }
+
+    // left expression should be hoge.length
+    let left = solve_parenthesized_expression(left)?;
+    let length_parent = get_length_node(&left)?;
+    // left expression should be the same as the object
+    if !is_same_reference(object, length_parent)? {
+        return None;
+    }
+
+    if right_list.len() == 1 {
+        // right expression should be integer
+        if let Some(number) =
+            get_integer_from_literal(&solve_parenthesized_expression(right_list[0].clone())?)
+        {
+            if number > 0 {
+                Some(AnyJsExpression::JsUnaryExpression(
+                    make::js_unary_expression(make::token(T![-]), right_list[0].clone()),
+                ))
+            } else {
+                None
+            }
+        } else {
+            Some(AnyJsExpression::JsUnaryExpression(
+                make::js_unary_expression(
+                    make::token(T![-]),
+                    overwrap_parentheses_expression(&right_list[0])?,
+                ),
+            ))
+        }
+    } else {
+        make_plus_binary_expression(right_list)
+    }
+}
+
+/// Is the node a child node of `delete`?
+fn is_within_delete_expression(node: &AnyJsExpression) -> Option<bool> {
+    node.syntax().parent()?.ancestors().find_map(|ancestor| {
+        if let Some(unary) = JsUnaryExpression::cast(ancestor.clone()) {
+            unary
+                .operator_token()
+                .ok()
+                .map(|token| token.kind() == T![delete])
+                .or(Some(false))
+        } else {
+            (!JsParenthesizedExpression::can_cast(ancestor.kind())).then_some(false)
+        }
+    })
+}
+
+fn make_number_literal(value: i64) -> AnyJsExpression {
+    AnyJsExpression::AnyJsLiteralExpression(AnyJsLiteralExpression::JsNumberLiteralExpression(
+        make::js_number_literal_expression(make::js_number_literal(value)),
+    ))
+}
+
+/// check if the node is a slice
+/// # Examples
+/// ```js
+///     .slice(0)[0]
+///     .slice(0, 1).pop(0)
+/// ```
+fn analyze_slice_element_access(node: &AnyJsExpression) -> Option<UseAtIndexState> {
+    if is_within_delete_expression(node).unwrap_or(false) {
+        return None;
+    }
+    // selector
+    let (selected_exp, extract_type): (AnyJsExpression, SliceExtractType) = match node {
+        // .pop() or .shift()
+        AnyJsExpression::JsCallExpression(call_exp) => {
+            let arg_length = call_exp.arguments().ok()?.args().into_iter().count();
+            if arg_length != 0 {
+                return None;
+            }
```

Comment:
Even simpler and more efficient (this requires to import `biome_rowan::AstSeparatedList`:
```suggestion
            let has_args = !call_exp.arguments().ok()?.args().is_empty();
            if has_args {
                return None;
            }
```